File: Modules/ServerTypes/Default/GovernateRow.ts
import { getLookup, getLookupAsync, fieldsProxy } from "@serenity-is/corelib";

export interface GovernateRow {
    GovernateId?: number;
    GovernateName?: string;
}

export abstract class GovernateRow {
    static readonly idProperty = 'GovernateId';
    static readonly nameProperty = 'GovernateName';
    static readonly localTextPrefix = 'Default.Governate';
    static readonly lookupKey = 'Default.Governate';

    /** @deprecated use getLookupAsync instead */
    static getLookup() { return getLookup<GovernateRow>('Default.Governate') }
    static async getLookupAsync() { return getLookupAsync<GovernateRow>('Default.Governate') }

    static readonly deletePermission = 'Administration:General';
    static readonly insertPermission = 'Administration:General';
    static readonly readPermission = 'Administration:General';
    static readonly updatePermission = 'Administration:General';

    static readonly Fields = fieldsProxy<GovernateRow>();
}

File: Modules/Default/Governate/GovernateRow.cs
namespace Serene2.Default;

[ConnectionKey("Default"), Module("Default"), TableName("Governate")]
[DisplayName("Governate"), InstanceName("Governate")]
[ReadPermission("Administration:General")]
[ModifyPermission("Administration:General")]
[ServiceLookupPermission("Administration:General")]
[LookupScript("Default.Governate",Permission ="?") ]
public sealed class GovernateRow : Row<GovernateRow.RowFields>, IIdRow, INameRow
{
    [DisplayName("Governate Id"), PrimaryKey, IdProperty]
    public int? GovernateId { get => fields.GovernateId[this]; set => fields.GovernateId[this] = value; }

    [DisplayName("Governate Name"), Size(10), NotNull, QuickSearch, NameProperty]
    public string GovernateName { get => fields.GovernateName[this]; set => fields.GovernateName[this] = value; }

    public class RowFields : RowFieldsBase
    {
        public Int32Field GovernateId;
        public StringField GovernateName;
    }
}